fs/layer/node.go

```diff
@@ -953,8 +955,30 @@ func defaultStatfs(stat *fuse.StatfsOut) {
 	stat.Files = 0 // dummy
 	stat.Ffree = 0
 	stat.Bsize = blockSize
-	stat.NameLen = 1<<32 - 1
+	// just make it 255 bytes as same as other filesystems
+	// https://en.wikipedia.org/wiki/Comparison_of_file_systems#Limits
+	stat.NameLen = 255
```

Comment:
xref for other reviewers if it's helpful: https://github.com/containerd/stargz-snapshotter/pull/2024